bundles/org.openhab.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioServlet.java

```diff
@@ -143,16 +146,20 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
     }
 
     private synchronized void removeTimedOutStreams() {
-        for (String streamId : multiTimeStreams.keySet()) {
-            if (streamTimeouts.get(streamId) < System.nanoTime()) {
-                // the stream has expired, we need to remove it!
-                FixedLengthAudioStream stream = multiTimeStreams.remove(streamId);
-                streamTimeouts.remove(streamId);
-                IOUtils.closeQuietly(stream);
-                stream = null;
-                logger.debug("Removed timed out stream {}", streamId);
+        // Build list of expired streams.
+        final List<String> toRemove = new LinkedList<>();
+        for (Entry<String, Long> entry : streamTimeouts.entrySet()) {
+            if (entry.getValue() < System.nanoTime()) {
+                toRemove.add(entry.getKey());
             }
         }
+        toRemove.forEach(streamId -> {
+            // the stream has expired, we need to remove it!
+            final FixedLengthAudioStream stream = multiTimeStreams.remove(streamId);
+            streamTimeouts.remove(streamId);
+            IOUtils.closeQuietly(stream);
```

Comment:
@maggu2810 I think in this line you can find a piece of the puzzle to answer #532.

Reply:
Not really. While reading this implementation there are much more things that confused me. I answered to #532